src/components/text-input/daikin-text-input.ts

```diff
@@ -129,37 +191,60 @@ export class DaikinTextInput extends LitElement {
   autocomplete?: HTMLInputElement["autocomplete"];
 
   /**
-   * Error state. Ignored if the `disabled` is `true`.
+   * Set a icon in the left of label
    */
-  @property({ type: Boolean, reflect: true })
-  error = false;
+  @property({ type: String })
+  leftIcon: IconType | null = null;
+
+  /**
+   * Set a icon in the right of label
+   */
+  @property({ type: String })
+  rightIcon: IconType | null = null;
 
   private _handleInput(e: InputEvent): void {
     this.value = (e.target as HTMLInputElement).value;
     this._internals.setFormValue(this.value);
   }
 
   override render() {
-    const textInputInputClassName = cvaInput({
-      variant: !this.disabled && this.error ? "error" : "normal",
-    });
-
-    return html`<input
-      class=${textInputInputClassName}
-      type=${this.type}
-      value=${this.value}
-      placeholder=${this.placeholder}
-      name=${ifDefined(this.name)}
-      maxlength=${ifDefined(this.maxlength)}
-      autocomplete=${
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any -- workaround lit-analyzer checking
-        ifDefined(this.autocomplete as any)
-      }
-      ?disabled=${this.disabled}
-      ?readonly=${this.readonly}
-      @change=${(e: Event) => this.dispatchEvent(new Event("change", e))}
-      @input=${this._handleInput}
-    />`;
+    const createIcon = (position: "left" | "right", icon: IconType | null) =>
+      icon
+        ? html`<div
+            class=${cvaIconContainer({
+              disabled: this.disabled,
+              position,
+            })}
+          >
+            <daikin-icon icon=${icon} size="m" color="current"></daikin-icon>
+          </div>`
+        : nothing;
+
+    return html`<div class="w-full h-full relative">
+      <input
+        class=${cvaInput({
+          error: !this.disabled && this.error,
+          leftIcon: !!this.leftIcon,
+          rightIcon: !!this.rightIcon,
+        })}
+        type=${this.type}
+        value=${this.value}
+        placeholder=${this.placeholder}
+        name=${ifDefined(this.name)}
+        maxlength=${ifDefined(this.maxlength)}
+        autocomplete=${
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any -- workaround lit-analyzer checking
+          ifDefined(this.autocomplete as any)
+        }
+        ?disabled=${this.disabled}
+        ?readonly=${this.readonly}
+        aria-required=${this.required}
```

Comment:
I think we should use `required` attr.

```suggestion
        ?required=${this.required}
```

Reply:
@poetrainy 
Doesn't seem to be fixed.

Please add a reaction after the change has actually been made. Or, if you think that this change is not suitable, please leave a comment.